src/groupmodel.cpp

```diff
@@ -280,65 +272,55 @@ QVariant GroupModel::data(const QModelIndex &index, int role) const
         return group->endTime().toLocalTime().date();
     }
 
-    int column = index.column();
-    if (role >= BaseRole) {
-        column = role - BaseRole;
-        role = Qt::DisplayRole;
-    }
-    
-    if (role != Qt::DisplayRole) {
-        return QVariant();
-    }
-
     QVariant var;
-    switch (column) {
-        case GroupId:
+    switch (role) {
```

Comment:
Also here it seems a little strange to do first if comparisons to `role` and then have a switch. Maybe these could be both in the same switch.